source/common/upstream/load_balancer_impl.cc

```diff
@@ -145,65 +147,66 @@ bool LoadBalancerUtility::isGlobalPanic(const HostSet& host_set, Runtime::Loader
+  // At this point it's guaranteed to be at least 2 localitys.
+  size_t number_of_localitys = host_set_.healthyHostsPerLocality().size();
 
-  ASSERT(number_of_zones >= 2U);
-  ASSERT(local_host_set_->healthyHostsPerZone().size() == host_set_.healthyHostsPerZone().size());
+  ASSERT(number_of_localitys >= 2U);
+  ASSERT(local_host_set_->healthyHostsPerLocality().size() ==
+         host_set_.healthyHostsPerLocality().size());
 
-  // Try to push all of the requests to the same zone first.
-  if (zone_routing_state_ == ZoneRoutingState::ZoneDirect) {
+  // Try to push all of the requests to the same locality first.
+  if (locality_routing_state_ == LocalityRoutingState::LocalityDirect) {
     stats_.lb_zone_routing_all_directly_.inc();
-    return host_set_.healthyHostsPerZone()[0];
+    return host_set_.healthyHostsPerLocality()[0];
   }
 
-  ASSERT(zone_routing_state_ == ZoneRoutingState::ZoneResidual);
+  ASSERT(locality_routing_state_ == LocalityRoutingState::LocalityResidual);
 
-  // If we cannot route all requests to the same zone, we already calculated how much we can
-  // push to the local zone, check if we can push to local zone on current iteration.
+  // If we cannot route all requests to the same locality, we already calculated how much we can
+  // push to the local locality, check if we can push to local locality on current iteration.
   if (random_.random() % 10000 < local_percent_to_route_) {
     stats_.lb_zone_routing_sampled_.inc();
-    return host_set_.healthyHostsPerZone()[0];
+    return host_set_.healthyHostsPerLocality()[0];
   }
 
-  // At this point we must route cross zone as we cannot route to the local zone.
+  // At this point we must route cross locality as we cannot route to the local locality.
   stats_.lb_zone_routing_cross_zone_.inc();
 
   // This is *extremely* unlikely but possible due to rounding errors when calculating
-  // zone percentages. In this case just select random zone.
-  if (residual_capacity_[number_of_zones - 1] == 0) {
+  // locality percentages. In this case just select random locality.
+  if (residual_capacity_[number_of_localitys - 1] == 0) {
     stats_.lb_zone_no_capacity_left_.inc();
-    return host_set_.healthyHostsPerZone()[random_.random() % number_of_zones];
+    return host_set_.healthyHostsPerLocality()[random_.random() % number_of_localitys];
   }
 
-  // Random sampling to select specific zone for cross zone traffic based on the additional
-  // capacity in zones.
-  uint64_t threshold = random_.random() % residual_capacity_[number_of_zones - 1];
+  // Random sampling to select specific locality for cross locality traffic based on the additional
+  // capacity in localitys.
+  uint64_t threshold = random_.random() % residual_capacity_[number_of_localitys - 1];
 
-  // This potentially can be optimized to be O(log(N)) where N is the number of zones.
+  // This potentially can be optimized to be O(log(N)) where N is the number of localitys.
```

Comment:
let me guess: emacs query-replace ? /zone/locality/ ? ;)

Reply:
Yes, tree-wide search/replacing, will fix tomorrow, thanks for picking up these.